Add postgis reflection query

othercorey opened the pull request:
This is unused until we can decide how to handle the various versions of POINT.

Reverted the column regex now that we don't need to parse complex types.

othercorey committed:
Add postgis reflection query

othercorey commented on src/Database/Schema/PostgresSchemaDialect.php:
```diff
@@ -111,6 +111,27 @@ private function describeColumnQuery(): string
+
+        $columns = $this->_driver->execute($sql, [$table, $schema, $catalog])->fetchAll('assoc');
+
+        return array_combine(array_column($columns, 'name'), $columns);
```
I think mysql has the same table structure for their geospatial types.

dereuromark commented:
Small CS fail.

markstory reviewed (approved):
Looks good to me

dereuromark commented:
@markstory 
You approved this.

> Unused variable $postgisColumns                  

The core question is: how should the PostGIS metadata (geometry/geography type and SRID) be represented in CakePHP's schema reflection? Once that's decided, the $postgisColumns data needs to be merged into the column info being returned.          
Currently it seems incomplete.

dereuromark committed:
Use PostGIS metadata in column schema reflection

Add geometryType and srid properties to geometry/geography columns
using the data fetched from PostGIS metadata tables.

dereuromark committed:
Update test expectations for PostGIS metadata

Now that we properly reflect geometryType and srid from PostGIS
metadata tables, update the test to expect the actual values.

dereuromark committed:
Add geometryType support to schema reflection

- Add geometryType to Column class (constructor, getter/setter, toArray)
- Add geometryType to $_columnExtras for geospatial types in TableSchema
- Add geography type to $_columnExtras (was missing)
- Convert PostGIS type to title case for consistency (GEOMETRY -> Geometry)
